File: src/services/schema-export/tests/models/sample/objc/SampleTypes.m
#import "RealmModels.h"

@implementation SampleTypes

+ (NSArray<NSString *> *)requiredProperties {
    return @[
        @"listOfStrings",
        @"listOfObjects",
    ];
}

+ (NSString *)primaryKey {
    return @"primary";
}

+ (NSArray<NSString *> *)indexedProperties {
    return @[
        @"indexedInt",
    ];
}

@end

File: src/services/schema-export/tests/models/sample/swift/SampleTypes.swift
import Foundation
import RealmSwift

class SampleTypes: Object {
    @objc dynamic var primary: Int = 0
    let ArrayFloatValue = RealmOptional<Float>()
    let listOfStrings = List<String>()
    let listOfOptionalDates = List<Date?>()
    @objc dynamic var indexedInt: Int = 0
    @objc dynamic var linkToObject: SampleTypes?
    let listOfObjects = List<SampleTypes>()

    override static func primaryKey() -> String? {
        return "primary"
    }

    override static func indexedProperties() -> [String] {
        return ["indexedInt"]
    }
}

